pkg/nsx/services/firewall.go

```diff
@@ -1033,13 +1034,31 @@ func (service *SecurityPolicyService) updatePeerExpressions(obj *v1alpha1.Securi
 	}
 	if peer.NamespaceSelector != nil {
 		if !mixedNsSelector {
+			if peer.NamespaceSelector.Size() == 0 {
```

Comment:
Do we need to check whether vmSelector and podSelector are nil here?

Reply:
It's not necessary to check here, since here logic is a branch for a non mixed criteria, which means both vmSelector and podSelector are nil. If either one is non nil, it will be a mixed criteria and go to code line 1062.